### PR TITLE
Improved file handling for un-archived files

### PR DESCRIPTION
**Pull request by danhedron**
This fixes the following:
- style fixes & better indentation
- Improve some naming
- Implement case-insensitive lookup of files in the game directory
- Load files from explicit paths instead of finding them using just their filename
- Remove some dead code
- Load IDE files at the correct time

### CODE REVIEW

**Comment by JayFoxRox** [comment]
This would fix #154 .

I think #176 also fits well with this one.

**Comment by danhedron** [comment]
- remove the FileFuture from this, since it's not needed at all
- rebase to master

File: rwengine/src/engine/GameData.cpp [hunk 1]
```diff
@@ -49,10 +38,15 @@ GameData::~GameData()
...
 {
+	index.indexGameDirectory(datpath);
 	index.indexTree(datpath);
-	
-	parseDAT(datpath+"/data/default.dat");
-	parseDAT(datpath+"/data/gta3.dat");
+
+	// Initalize all the archives.
+	loadIMG("models/gta3.img");
+	loadIMG("anim/cuts.img");
```
Comment (danhedron) [inline rwengine/src/engine/GameData.cpp]: Needs a note that this shouldn't be considered for loading files in the same way as gta3.img

File: rwengine/src/engine/GameData.cpp [hunk 2]
```diff
@@ -62,102 +56,81 @@ void GameData::load()
...
-						if(texpath[t] == '\\') {
-							texpath[t] = '/';
-						}
-					}
-					std::string texname = texpath.substr(texpath.find_last_of("/")+1);
-					loadTXD(texname);
-				}
+				auto path = line.substr(space+1);
+				auto systempath = index.findFilePath(path).native();
+				loadIDE(systempath);
```
Comment (danhedron) [inline rwengine/src/engine/GameData.cpp]: This resolve-before-call style is at different from the way the other loadX functions are implemented. Should be consistent.

**Comment by danhedron** [comment]
I've reviewed this a few times myself. There may be one or two things worth improving but there are more changes coming soon and this is holding up another PR. Merging it now and anything else can be fixed in another PR.

**Pull request merged**